src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/resource/AuditResourceTest.kt

```diff
@@ -143,16 +143,82 @@ class AuditResourceTest : NoQueueListenerIntegrationTest() {
     @ParameterizedTest
     @MethodSource("secureEndpoints")
     internal fun `satisfies the correct role and scope`(uri: String) {
+      doNothing().whenever(auditService).sendAuditEvent(any())
+
       webTestClient.post()
         .uri(uri)
         .headers(setAuthorisation(roles = listOf("ROLE_AUDIT"), scopes = listOf("write")))
         .body(BodyInserters.fromValue(AuditEvent(what = "secureEndpointCheck")))
         .exchange()
         .expectStatus().isAccepted
 
+      verify(auditService).sendAuditEvent(any())
+    }
+  }
+
+  @Nested
+  inner class traceParentHeader {
+    @Test
+    internal fun `no traceparent header`() {
       doNothing().whenever(auditService).sendAuditEvent(any())
 
-      verify(auditService).sendAuditEvent(any())
+      val auditEvent = AuditEvent(what = "traceparent empty event")
+      webTestClient.post()
+        .uri("/audit")
+        .headers(setAuthorisation(roles = listOf("ROLE_AUDIT"), scopes = listOf("write")))
+        .body(BodyInserters.fromValue(auditEvent))
+        .exchange()
+        .expectStatus().isAccepted
+
+      verify(auditService).sendAuditEvent(auditEvent)
+    }
+
+    @Test
+    internal fun `invalid traceparent header`() {
+      doNothing().whenever(auditService).sendAuditEvent(any())
+
+      val auditEvent = AuditEvent(what = "traceparent invalid event")
+      webTestClient.post()
+        .uri("/audit")
+        .headers(setAuthorisation(roles = listOf("ROLE_AUDIT"), scopes = listOf("write")))
+        .header("traceparent", "3d6cb11f59448eb9d50a7f1e5237")
+        .body(BodyInserters.fromValue(auditEvent))
+        .exchange()
+        .expectStatus().isAccepted
+
+      verify(auditService).sendAuditEvent(auditEvent)
+    }
+
+    @Test
+    internal fun `valid traceparent header`() {
+      doNothing().whenever(auditService).sendAuditEvent(any())
+
+      val auditEvent = AuditEvent(what = "traceparent valid event", operationId = "8c3d6cb11f59448eb9d50a7f1e523748")
+      webTestClient.post()
+        .uri("/audit")
+        .headers(setAuthorisation(roles = listOf("ROLE_AUDIT"), scopes = listOf("write")))
+        .header("traceparent", "00-8c3d6cb11f59448eb9d50a7f1e523748-adf0569620934d76-01")
+        .body(BodyInserters.fromValue(auditEvent))
```

Comment:
these tests are working because the `operationId` is always being passed in in the body, so it never hits the code that then copies it from the request header

Reply:
maybe if you passed in the json here instead you can test the hydration too and will also prevent accidentally changing the signature of the body by updating the `AuditEvent` class in the future

Reply:
added new test

Reply:
Can see your point re the json.  Once the messagingTemplate is mocked out it will mean the doNothing() method call is not required and so will be a simpler test.  Will update in another PR